src/lib/services/tx/index.ts

```diff
@@ -1,85 +1,100 @@
-import type {
-  QueryFunctionContext,
-  UseQueryOptions,
-  UseQueryResult,
-} from "@tanstack/react-query";
 import { useQuery } from "@tanstack/react-query";
+import type { UseQueryOptions, UseQueryResult } from "@tanstack/react-query";
 import { useCallback } from "react";
 
+import type {
+  AccountTxsResponse,
+  BlockTxsResponse,
+  TxData,
+  TxsResponse,
+} from "../types/tx";
 import {
   CELATONE_QUERY_KEYS,
   useBaseApiRoute,
   useCelatoneApp,
   useInitia,
+  useLcdEndpoint,
   useMoveConfig,
+  useTierConfig,
   useWasmConfig,
 } from "lib/app-provider";
-import { getTxsByPoolIdPagination, getTxsCountByPoolId } from "lib/query";
-import { createQueryFnWithTimeout } from "lib/query-utils";
-import type {
-  BechAddr,
-  Message,
-  Option,
-  PoolTxFilter,
-  Transaction,
-  TxFilters,
-} from "lib/types";
-import { ActionMsgType, MsgFurtherAction } from "lib/types";
-import {
-  extractTxLogs,
-  isTxHash,
-  parseDate,
-  parseTxHash,
-  snakeToCamel,
-} from "lib/utils";
+import type { BechAddr, Option, TxFilters } from "lib/types";
+import { extractTxLogs, isTxHash, parseDateOpt } from "lib/utils";
 
-import { usePoolTxExpression } from "./expression";
-import type {
-  AccountTxsResponse,
-  BlockTxsResponse,
-  TxResponse,
-  TxsResponse,
-} from "./tx";
 import {
+  getTxData,
   getTxs,
   getTxsByAddress,
   getTxsByBlockHeight,
   getTxsCountByAddress,
-  queryTxData,
-} from "./tx";
-
-export interface TxData extends TxResponse {
-  chainId: string;
-  isTxFailed: boolean;
-}
+} from "./api";
+import { getTxDataLcd } from "./lcd";
 
 export const useTxData = (
   txHash: Option<string>,
   enabled = true
 ): UseQueryResult<TxData> => {
   const { currentChainId } = useCelatoneApp();
-  const txsApiRoute = useBaseApiRoute("txs");
+  const tier = useTierConfig();
+  const apiEndpoint = useBaseApiRoute("txs");
+  const lcdEndpoint = useLcdEndpoint();
+
```

Comment:
Let's declare `endpoint` variable and use it in `getTxData`, `getTxDataLcd` functions.